tests/integration/test_secrets.py

```diff
@@ -84,3 +82,20 @@ async def test_happy_delete(
         response = await client.get(TEST_URL, headers=HEADERS)
         assert response.status_code == 200
         assert response.json() == []
+
+
+async def test_nonexistent_vault_path(vault: VaultFixture):
+    """Ensure both endpoint return the right response when `vault_path` doesn't exist."""
+    vault_fixture_config = vault.config
+    config = get_config(sources=[vault_fixture_config])
+
+    async with (
+        prepare_rest_app(config=config) as app,
+        AsyncTestClient(app=app) as client,
+    ):
+        response = await client.get("/secrets/doesnotexist", headers=HEADERS)
+        assert response.status_code == 200
```

Comment:
Why is this a 200?

Reply:
I think because you cannot distinguish the invalid path from an empty path, it returns an empty list in either case.